### ISSUE TITLE
Some question about the value of action product by policy network

### ISSUE THREAD

**Issue by Jeremy-07** [nooralahzadeh/DSNER-pytorch#5]
Hello, after 15 rounds of running your code, I found that the value of a was approaching 1, which led to the selection of all distant supervision data. This is obviously unreasonable, is there any way to solve it?

**Comment by nooralahzadeh** [comment]
The coder is the PyTorch implementation of  the following repository: https://github.com/rainarch/DSNER
The issue that you mention is already addressed in the following paper and also proposed a solution:
https://www.aclweb.org/anthology/D19-6125/

**Comment by PengShi27** [comment]
I read the article you mentioned in the comments section. I redid your experiment according to the method you proposed, but I met several problems in the process of redoing the experiment.The first problem is that according to algorithm 1, the NER model must be pre-trained first, but your article did not give the specific number of training rounds. The second problem is that this article uses the difference of F score as reward, and then I according to this method to update the selector, but aas the training went on, I found that the value of the reward has been volatile, if there is no convergence, the value of the reward means reward didn't good to update the parameters of the selector, choose reasonable examples so as to promote the selector.I would appreciate it very much if you could give me some reasonable advice.

**Comment by Jeremy-07** [comment]
Hello, I have read the article you mentioned in the comment section, in which you used the difference of F score as a reward, which is very interesting. But I tried to repeat your experiment and found several problems. The first problem is that the article mentions that the NER model should be pre-trained before retraining, but you don't seem to have given a specific number of pre-training rounds in the article. The second problem is that when we update the selector with the difference of F score as a reward, the reward value is oscillating and doesn't converge, which means that the selector isn't picking the instance very well, thus leading to the poor performance of the NER model. I would appreciate it if you could give me some advice.
